dashboard/app/models/teacher_feedback.rb

```diff
@@ -57,23 +60,45 @@ def get_script_level
     script_level
   end
 
-  def self.get_student_level_feedback(student_id, level_id, teacher_id, script_id)
+  def self.get_student_level_feedback_by_teacher(student_id, level_id, script_id, teacher_id)
     where(
       student_id: student_id,
       level_id: level_id,
-      teacher_id: teacher_id,
-      script_id: script_id
+      script_id: script_id,
+      teacher_id: teacher_id
     ).latest
   end
 
-  def self.get_all_feedback_for_section(student_ids, level_ids, teacher_id)
-    find(
-      where(
-        student_id: student_ids,
-        level_id: level_ids,
-        teacher_id: teacher_id
-      ).group([:student_id, :level_id]).pluck('MAX(teacher_feedbacks.id)')
-    )
+  def self.get_all_student_level_feedback(student_id, level_id, script_id)
+    where(
+      student_id: params.require(:student_id),
+      level_id: params.require(:level_id),
+      script_id: params.require(:script_id)
+    ).latest_per_teacher
+  end
+
+  def self.student_has_feedback(student_id)
+    where(student_id: student_id).count > 0
+  end
+
+  def self.get_student_unseen_feedback_count(student_id)
+    authorized_unseen_feedbacks = where(
+      student_id: current_user.id,
+      seen_on_feedback_page_at: nil,
+      student_first_visited_at: nil
+    ).select do |feedback|
+      feedback.teacher.authorized_teacher?
```

Comment:
This is a direct refactor of the previous query. Currently only authorized teachers can leave feedback, so this seems unnecessary. Should we remove?